src/main/java/com/google/sps/servlets/ProfileInfoServlet.java

```diff
@@ -33,18 +34,29 @@
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
 
     //   TODO: include datastore for profile's and retrieve profile data from datastore
-    // initialize Profile with dummy data
-    String name = "John Doe";
-    ArrayList<String> projectHistory = new ArrayList<String>();
-    projectHistory.add("Project 1");
-    projectHistory.add("Project 2");
-    ArrayList<String> reviews = new ArrayList<String>();
-    reviews.add("Very reliable!");
-    reviews.add("Great communicator and 100% reliable!");
-    HashMap<String, String> contactInfo = new HashMap<String, String>();
-    contactInfo.put("Mobile", "(123) 456-7809");
-    contactInfo.put("Email", "johndoe@gmail.com");
-    Profile profile1 = new Profile(name, projectHistory, reviews, contactInfo);
+    
+    // parse email for name
+    String email = request.getParameter("user_id");
+    int endIndex;
+    String name;
+    if(email != null && email.contains("@")){
+        endIndex = email.indexOf("@");
+        name = email.substring(0, endIndex);
+        System.out.println("inside");
+    }
+    else{
+        name = "John Doe";
```

Comment:
I know this is a generic name but maybe using "No Username" is less misleading.

Reply:
good suggestion!

Reply:
+1